### ISSUE TITLE
To compile Cython sources code both `cython` and `c` need to be passed to `project()`

### ISSUE THREAD

**Issue by dnicolodi** [mesonbuild/meson#11389]
I don't know if this is expected because Cython is a a transpiler to C or C++, but using Cython requires passing both `cython` and `c` to `project()` otherwise Meson complains with `no toolchain found`. The strange thing is that this happens only on some platforms (some Linux distributions but not others, for example).

**Comment by eli-schwartz** [comment]
This error string is only ever emitted by `dependency()` when it is using a `method: 'system'` lookup that tries to access a `self.clib_compiler` in order to internally run find_library or has_header or similar. You should therefore be able to get that error message regardless of whether the project uses cython at all.

**Comment by dnicolodi** [comment]
This is the whole `meson.build`:
```meson
project('complex', 'c', 'cython', version: '1.0.0')

py = import('python').find_installation()

install_subdir('complex', install_dir: py.get_install_dir(pure: false))

py.extension_module('test', 'test.pyx', install: true, subdir: 'complex')
```
If I remove `c` from the `project()` call, I get the `no toolchain found` error. Example: https://cirrus-ci.com/task/5918684011560960

**Comment by eli-schwartz** [comment]
Aha, yes. That happens because:
```
Did not find pkg-config by name 'pkg-config'
Found Pkg-config: NO
../../../../cirrus-ci-build/tests/packages/complex/meson.build:11:3: ERROR: no toolchain found
```

So pkg-config was not found, and the python module's find_installation + dependency handling attempted to find libpython using sysconfig + the clib_compiler.

**Comment by dnicolodi** [comment]
I see. It is a bit unexpected.

**Comment by eli-schwartz** [comment]
Yes, but we used to simply crash instead. :D Now the dependencies are just not found, which in some cases are mandatory dependencies.

The question is, how this plays together with cython... do we not need to specify both cython and c just to ensure the transpiled files can then be compiled?

**Comment by dnicolodi** [comment]
Dunno, on some platforms it worked without specifying `c`. Maybe the less surprising thing would be to error out if someone specifies `cython` without adding `c` or `cpp`, if it is reasonable to implement.

**Comment by rgommers** [comment]
A user ran into this in https://github.com/mesonbuild/meson-python/discussions/558.

> Maybe the less surprising thing would be to error out if someone specifies `cython` without adding `c` or `cpp`, if it is reasonable to implement.

This seems like a good solution indeed. Automatically adding `'c'` would be a bit more questionable, since it's very well possible that the user only wants `'cpp'`.

**Comment by JamBalaya56562** [comment]
Thank you for the report. This is fixed in Meson 1.11: when `cython` is passed to `project()`, Meson now automatically enables the Cython backing language (C by default, or whatever `cython_language` is set to), so you no longer need to list `c` (or `cpp`) explicitly.

```python
# mesonbuild/interpreter/interpreter.py
_lang = self.coredata.optstore.get_pending_value(
    OptionKey('cython_language', None, for_machine), 'c')
...
FeatureNew.single_use(f'Adding Cython language without {cython_lang}', '1.11', ...)
```

**Verification:** the release note ["Cython no longer requires explicitly enabling C or C++"](https://mesonbuild.com/Release-notes-for-1.11.0.html), and the auto-add logic (guarded by `FeatureNew('Adding Cython language without <lang>', '1.11')`) in `mesonbuild/interpreter/interpreter.py`.

With Meson 1.11 or newer the extra `c` is no longer required, so this looks safe to close. Thanks for raising it.
